test/nl/hannahsten/texifyidea/completion/LatexGlossaryCompletionTest.kt

```diff
@@ -0,0 +1,74 @@
+package nl.hannahsten.texifyidea.completion
+
+import com.intellij.codeInsight.completion.CompletionType
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.codeInsight.lookup.LookupElementPresentation
+import com.intellij.testFramework.fixtures.BasePlatformTestCase
+import nl.hannahsten.texifyidea.file.LatexFileType
+import org.junit.Test
+
+class LatexGlossaryCompletionTest : BasePlatformTestCase() {
+
+    override fun getTestDataPath(): String {
+        return "test/resources/completion/glossary"
+    }
+
+    private fun validateLookupElement(
+        lookup: String,
+        short: String,
+        description: String,
+        elements: Array<LookupElement>
+    ) {
+        val element = elements.singleOrNull { l -> l.lookupString == lookup }
+        assertNotNull("$lookup glossary entry not found", element)
+        val presentation = LookupElementPresentation()
+        element!!.renderElement(presentation)
+        assertTrue("$lookup short description not found", presentation.typeText?.contains(short) ?: false)
+        assertTrue("$lookup long description not found", presentation.tailText?.contains(description) ?: false)
+    }
+
+    @Test
+    fun testCompleteGlossaryCommandEntries() {
+        // given
+        myFixture.configureByFiles("${getTestName(false)}.tex")
+
+        // when
+        val result = myFixture.complete(CompletionType.BASIC)
+
+        // then
```

Comment:
As an aside, this reminds me of the Spek framework, do you know it? It has Gherkin syntax, I kinda liked it. Though I think I never migrated to Spek 2 because I preferred Spek 1 syntax.